src/canvas/Point.tsx

```diff
@@ -42,10 +40,21 @@ export class Point implements IPoint {
         return minDistance;
     };
 
-    computeMiddlePoint = (point2: Point): Point => {
-        const middleX = (this.getX() + point2.getX()) / 2;
-        const middleY = (this.getY() + point2.getY()) / 2;
+    static computeMiddlePoint = (p1: Point, p2: Point): Point => {
+        const middleX = (p1.x + p2.x) / 2;
+        const middleY = (p1.y + p2.y) / 2;
         
         return new Point(middleX, middleY);
     };
+
+    // generates points on the line formed by all points that are equidistant 
+    // from points p1 and p2.
+    // at t=0, is returns the point in the middle of this point and the other point.
+    static getPointOnMidline = (p1: Point, p2: Point, t: number): Point => {
```

Comment:
I'm using vectors in the calculations, but from the point of vue of someone using the function, they don't know that. The method takes 2 points and returns a point, so I feel like it would not fit inside the vector class

Reply:
Yeah, I see what you mean. I was just thinking that the idea of a midline seems specific to the curve calculation. Is it a more generic concept?

Reply:
I do think of it as a more generic concept. I don't know if we'll reuse it anywhere else though.

Reply:
As discussed in 8/13 meeting, I'll leave it in Point.